### PR TITLE
[inductor] move dynamic shapes tests into a new file

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94971

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by nkaretnikov** [comment]
This is based on master, which is probably why there are so many failures, will rebase on top of viable/strict once my previous inductor PR is part of that

**Review by ezyang — APPROVED** [review]

File: test/inductor/test_torchinductor_dynamic_shapes.py [hunk 1]
```diff
@@ -8,7 +11,10 @@
 )
 from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
 
-from .test_torchinductor import (
+# Make the helper files in test/ importable
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
```
Comment (nkaretnikov) [inline test/inductor/test_torchinductor_dynamic_shapes.py:16]: Same pattern as in the other tests to make test_torchinductor importable by pytest and python

File: test/inductor/test_torchinductor_dynamic_shapes.py [hunk 2]
```diff
@@ -21,6 +25,16 @@
...
 
+if IS_WINDOWS and IS_CI:
+    sys.stderr.write(
+        "Windows CI does not have necessary dependencies for test_torchinductor_dynamic_shapes yet\n"
+    )
+    if __name__ == "__main__":
+        sys.exit(0)
+    raise unittest.SkipTest("requires sympy/functorch/filelock")
+
+importlib.import_module("filelock")
```
Comment (nkaretnikov) [inline test/inductor/test_torchinductor_dynamic_shapes.py:36]: Same as in test_torchinductor

**Pull request closed without merging**